tests/Aspire.Hosting.Tests/Redis/AddRedisTests.cs

```diff
@@ -164,4 +164,105 @@ public async Task MultipleRedisInstanceProducesCorrectRedisHostsVariable()
+        var redis = appBuilder.AddRedis("myRedis");
+        if (isReadOnly.HasValue)
+        {
+            redis.WithDataVolume(isReadOnly: isReadOnly.Value);
+        }
+        else
+        {
+            redis.WithDataVolume();
+        }
+
+        var volumeAnnotation = redis.Resource.Annotations.OfType<ContainerMountAnnotation>().Single();
+
+        Assert.Equal("myRedis-data", volumeAnnotation.Source);
+        Assert.Equal("/data", volumeAnnotation.Target);
+        Assert.Equal(ContainerMountType.Volume, volumeAnnotation.Type);
+        Assert.Equal(isReadOnly ?? false, volumeAnnotation.IsReadOnly);
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void WithDataBindMountAddsMountAnnotation(bool? isReadOnly)
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var redis = appBuilder.AddRedis("myRedis");
+        if (isReadOnly.HasValue)
+        {
+            redis.WithDataBindMount("mydata", isReadOnly: isReadOnly.Value);
+        }
+        else
+        {
+            redis.WithDataBindMount("mydata");
+        }
+
+        var volumeAnnotation = redis.Resource.Annotations.OfType<ContainerMountAnnotation>().Single();
+
+        Assert.Equal("mydata", volumeAnnotation.Source);
+        Assert.Equal("/data", volumeAnnotation.Target);
+        Assert.Equal(ContainerMountType.BindMount, volumeAnnotation.Type);
+        Assert.Equal(isReadOnly ?? false, volumeAnnotation.IsReadOnly);
+    }
+
+    [Fact]
+    public void WithDataVolumeAddsPersistenceAnnotation()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var redis = appBuilder.AddRedis("myRedis")
+                              .WithDataVolume();
+
+        var persistenceAnnotation = redis.Resource.Annotations.OfType<RedisPersistenceCommandLineArgsCallbackAnnotation>().Single();
+
+        Assert.Equal(TimeSpan.FromSeconds(60), persistenceAnnotation.Interval);
+        Assert.Equal(1, persistenceAnnotation.KeysChangedThreshold);
+    }
+
+    [Fact]
+    public void WithDataBindMountAddsPersistenceAnnotation()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var redis = appBuilder.AddRedis("myRedis")
+                              .WithDataBindMount("myredisdata");
+
+        var persistenceAnnotation = redis.Resource.Annotations.OfType<RedisPersistenceCommandLineArgsCallbackAnnotation>().Single();
+
+        Assert.Equal(TimeSpan.FromSeconds(60), persistenceAnnotation.Interval);
+        Assert.Equal(1, persistenceAnnotation.KeysChangedThreshold);
+    }
+
+    [Fact]
+    public void WithPersistenceReplacesPreviousAnnotationInstances()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var redis = appBuilder.AddRedis("myRedis")
+                              .WithDataVolume()
+                              .WithPersistence(TimeSpan.FromSeconds(10), 2);
```

Comment:
Could this API design be confusing to developers? It would be easy for the developer to assume that either of these extension methods along should do what they want? A developer might see `WithPersistence` and thing that is what they want bit without either `WithDataVolme` or `WithDataBindMount` it isn't going to do anything.

I can see why the split but I do think it'll be confusing for folks.

Reply:
`WithPersistence` will work without a bind mount or volume, it will just persist to the container file system and be lost when the container is stopped. I can add more details to the doc comments to try and make it clearer. The alternative would be to remove the `WithPersistence` method and promote its arguments to `WithDataVolume` and `WithDataBindMount`.

Reply:
I updated the doc comments to make it clearer that calling `WithDataBindMount` or `WithDataVolume` is required to get the persistence to last across sessions.